File: common/programmers/k진수에서소수개수구하기/k진수에서소수개수구하기_고진석.java
class k진수에서소수개수구하기 {
    public int solution(int n, int k) {
    	int answer = 0;
    	char[] num = Integer.toString(n,k).toCharArray();
    	
    	String tmp = "";
    	for(int i = 0 ; i < num.length ; i++) {
    		if(num[i] != '0')
    			tmp += num[i];
    		else if(tmp.length() != 0) {
				if(isPrime(Long.parseLong(tmp)))
					answer++;
				tmp = "";
    		}
    	}
    	
    	if(tmp.length() != 0 && isPrime(Long.parseLong(tmp)))
    		answer++;
    	
        return answer;
    }
    
    public boolean isPrime(long num) {
    	if(num == 1)
    		return false;
    	for(int i = 2 ; i <= Math.sqrt(num) ; i++)
    		if(num % i == 0)
    			return false;
    	return true;
    }
}

File: common/programmers/k진수에서소수개수구하기/k진수에서소수개수구하기_고진석.py
import math

def solution(n, k):
    answer = 0

    N = convert(n,k)
    
    num = ''
    for c in N: 
        if c != '0':
            num += c
        elif num != '':
            if isPrime(num):
                answer += 1
            num = ''

    if num != '' and isPrime(num):
        answer += 1

    return answer

def convert(n, q):
    tmp = ''

    while n > 0:
        n, mod = divmod(n, q)
        tmp += str(mod)

    return tmp[::-1] 

def isPrime(x):
    x = int(x)
    if x == 1:
        return False
    for i in range(2, int(math.sqrt(x))+1):
        if x % i == 0:
            return False
    return True

print(solution(437674,3))